linux/devdoc/threadpool_linux_requirements.md

```diff
@@ -404,80 +404,89 @@ MOCKABLE_FUNCTION(, THREADPOOL_WORK_ITEM_HANDLE, threadpool_create_work_item, TH
+    - **SRS_THREADPOOL_LINUX_05_018: [** If reallocating the task array fails, `threadpool_schedule_work_item` shall fail by setting the return variable a `non-zero` value and break. **]**
 
-**S_R_S_THREADPOOL_LINUX_05_017: [** If reallocating the task array fails, `threadpool_schedule_work_item` shall fail and return a `NULL` value. **]**
+**SRS_THREADPOOL_LINUX_05_019: [** `threadpool_schedule_work_item` shall increment the `pending_work_item_count` by calling `interlocked_increment` and copy its address to `pending_work_item_count_ptr` into insert position in the task array. **]**
 
-**S_R_S_THREADPOOL_LINUX_05_018: [** `threadpool_schedule_work_item` shall increment the `pending_work_item_count` and copy its address to `pending_work_item_count_ptr` into insert position in the task array **]**
+**SRS_THREADPOOL_LINUX_05_020: [** `threadpool_schedule_work_item` shall copy the `work_function` and `work_function_context` from `threadpool_work_item` into insert position in the task array. **]**
 
-**S_R_S_THREADPOOL_LINUX_05_019: [** `threadpool_schedule_work_item` shall copy the `work_function` and `work_function_context` from `work_item` into insert position in the task array and assign `0` to the return variable to indicate success. **]**
+**SRS_THREADPOOL_LINUX_05_021: [** `threadpool_schedule_work_item` shall set the `task_state` to `TASK_WAITING` and then release the shared `SRW lock`. **]**
 
-**S_R_S_THREADPOOL_LINUX_05_020: [** `threadpool_schedule_work_item` shall set the `task_state` to `TASK_WAITING` and then release the shared SRW lock. **]**
+**SRS_THREADPOOL_LINUX_05_022: [** `threadpool_schedule_work_item` shall unblock the `threadpool` semaphore by calling `sem_post`. **]**
 
-**S_R_S_THREADPOOL_LINUX_05_021: [** `threadpool_schedule_work_item` shall unblock the `threadpool` semaphore by calling `sem_post`. **]**
+**SRS_THREADPOOL_LINUX_05_023: [** `threadpool_schedule_work_item` shall set the return variable to `0` to indicate success. **]**
 
-**S_R_S_THREADPOOL_LINUX_05_022: [** `threadpool_schedule_work_item` shall return `0` on success. **]**
+**SRS_THREADPOOL_LINUX_05_024: [** `threadpool_schedule_work_item` shall call `sm_exec_end`. **]**
 
-**S_R_S_THREADPOOL_LINUX_05_023: [** `threadpool_schedule_work_item` shall call `sm_exec_end`. **]**
+**SRS_THREADPOOL_LINUX_05_025: [** `threadpool_schedule_work_item` shall return with the contents of the value of the return variable. **]**
 
 ### threadpool_destroy_work_item
 
 ```c
-MOCKABLE_FUNCTION(, void, threadpool_destroy_work_item, THANDLE(THREADPOOL), threadpool, THREADPOOL_WORK_ITEM_HANDLE, work_item_context);
+MOCKABLE_FUNCTION(, void, threadpool_destroy_work_item, THANDLE(THREADPOOL), threadpool, THREADPOOL_WORK_ITEM_HANDLE, threadpool_work_item);
 ```
 
 `threadpool_destroy_work_item` Does nothing and a placeholder for WIN32 equivalent function stub
```

Comment:
>Does nothing and a placeholder for WIN32 equivalent function stub [](http://example.com/codeflow?start=31&length=65)

It is waiting for the scheduled work item to be complete before it destroys it now.